src/main/java/teammates/common/datatransfer/StudentAttributes.java

```diff
@@ -130,7 +129,7 @@ public boolean isRegistered() {
     }
 
     public String getRegistrationUrl() {
-        return new Url(Config.APP_URL + Const.ActionURIs.STUDENT_COURSE_JOIN_NEW)
+        return new Url(Const.ActionURIs.STUDENT_COURSE_JOIN_NEW)
```

Comment:
I think this change need not be part of this fix. We can create another one to make these changes. Being a very high priority issue, I think we should merge this pr soon.

For the question of whether we should make relative URLs the norm in the system, I am ok with it. In fact, if it works better than full URLs, we should move over. But since there is a real use-case for the full url, we should create a ```Full*Url()``` and ```Relative*Url()``` for clarity and minimise repetitive code. Also, this allows for unit testing the full URL as compared to testing it in ActionTests or browser tests